### PR TITLE
Some note/chart stuff

### PR DESCRIPTION
**Pull request by MaybeMaru**
Made strum note code a bit smaller
Fixed pixel strums framerate
Added sustain colors to chart editor
Added a difficulty selection to chart editor
Added a playback rate slider to chart editor
Fixed copy last section not working with the note and event checkbox stuff
Added lua functions getColorFromRgb and getDominantColor

### CODE REVIEW

**Review by MemeHoovy — APPROVED** [review]

**Review by Luisinhi010 — APPROVED** [review]

File: source/FunkinLua.hx [hunk 1]
```diff
@@ -1388,6 +1388,11 @@ class FunkinLua {
 			return Std.parseInt(color);
 		});
 
+		Lua_helper.add_callback(lua, "getColorFromRgb", function(r:Int, g:Int, b:Int) {
+			var color = FlxColor.fromRGB(r, g, b);
+			return color;
```
Comment (kingyomoma) [inline source/FunkinLua.hx:1393]: this would return a FlxColor object
`return FlxColor.toHexString(color, false);` would probably work instead

**Review by iym1337 — CHANGES_REQUESTED** [review]

**Comment by TinyGamesCoding** [comment]
this is radically radical

**Review by mosusg — APPROVED** [review]
Yes

**Review by TinyGamesCoding — APPROVED** [review]

**Comment by BlueColorsin** [comment]
Holy shit it is swag

**Review by BlueColorsin — APPROVED** [review]

**Review by iym1337 — APPROVED** [review]

**Review by Raltyro — APPROVED** [review]

**Review by JorosTheModderPH — APPROVED** [review]

**Review by MemeHoovy — APPROVED** [review]

**Review by lunacynne — APPROVED** [review]

**Review by TheKitBoi — APPROVED** [review]

**Review by rodney528 — APPROVED** [review]

**Review by Parsellol — APPROVED** [review]

**Comment by LEVIXDDLMAO** [comment]
dope

**Review by rodney528 — APPROVED** [review]

**Comment by MaybeMaru** [comment]
looking back this is a kinda trashy pr, soooo cya

**Pull request closed without merging**